Automattic/newspack-theme
# Header: Add simplied header option for subpages

### All Submissions:


### Changes proposed in this Pull Request:

This PR adds a theme option to use a very simple header on subpages. This header will only include the logo (no site title or tagline), a search toggle, and a menu toggle; the menu under the toggle includes all menus assigned, and any widgets assigned to the slide out widget area if populated and turned on.

Closes #877.

### How to test the changes in this Pull Request:

1. Apply the PR and run `npm run build`.
2. Navigate to Customize > Header Settings; the Header Options should now be split into three panels, with the existing one split over Appearance and Slide-out Sidebar, and a new one called Subpage Header:

![image](https://user-images.githubusercontent.com/177561/81025953-6658f580-8e2d-11ea-9e02-b0b737b853b6.png)

![image](https://user-images.githubusercontent.com/177561/81025964-6bb64000-8e2d-11ea-8ce8-368052da94de.png)

3. Check the box labelled  "Use simple header on subpages"; it should open an option to upload an Alternative Logo, but ignore that for now:

![image](https://user-images.githubusercontent.com/177561/81026348-ac628900-8e2e-11ea-810c-e6dd46333731.png)

4. View the homepage, and confirm that your header is displaying as expected:

![image](https://user-images.githubusercontent.com/177561/80547954-598c5b80-896e-11ea-84ba-781109fc737c.png)

5. View a regular single post, and confirm that you now have a simplified header (shown below with the default, solid background, and centered settings):

![image](https://user-images.githubusercontent.com/177561/80547541-5775cd00-896d-11ea-9ad2-6fbe7016d1e3.png)

![image](https://user-images.githubusercontent.com/177561/80547826-087c6780-896e-11ea-95aa-4cc113d77096.png)

![image](https://user-images.githubusercontent.com/177561/80547866-1b8f3780-896e-11ea-8ede-11144e22e5c6.png)

6. Try opening the menu and confirm that it contains your menus: 

![image](https://user-images.githubusercontent.com/177561/80547614-8be98900-896d-11ea-9bcd-bce8d2e22e42.png)

7. Check out a post with the 'behind' featured image style; confirm the header sits on top of the image (default and centred logo options shown below; background colours will not be applied to the header, even if assigned): 

![image](https://user-images.githubusercontent.com/177561/80547684-af143880-896d-11ea-8486-c094fa9994dc.png)

![image](https://user-images.githubusercontent.com/177561/80548150-0bc42300-896f-11ea-9c64-ac494aa285fb.png)

8. Check out a post with the 'beside' featured image styles; confirm the header elements sit over the solid colour:

![image](https://user-images.githubusercontent.com/177561/80547718-cb17da00-896d-11ea-968b-be6d146eadb2.png)

![image](https://user-images.githubusercontent.com/177561/80548178-21394d00-896f-11ea-8603-7cbbd8c75047.png)

9. Try scaling down the browser window -- the mobile header should kick in as usual, with a solid background and logo aligned left, regardless what your header settings are:

![image](https://user-images.githubusercontent.com/177561/80548535-02878600-8970-11ea-9b06-0c47f940f681.png)

10. Navigate back to Customize > Header Settings > Subpage Header, and upload an Alternative Logo. The use case here is if you have a dark logo that doesn't show up very well on the Behind and Beside featured image styles, this option can be used to replace it in those cases. So ideally the alternative logo will be a 'flipped' version of your site's real logo:

![image](https://user-images.githubusercontent.com/177561/81026428-efbcf780-8e2e-11ea-8b3f-c5e756dd37d3.png)

11. View the Featured Image Behind and Featured Image beside posts again; confirm the flipped logo is used:

![image](https://user-images.githubusercontent.com/177561/81029098-9bb71080-8e38-11ea-97fa-3ea43159043c.png)

![image](https://user-images.githubusercontent.com/177561/81029142-bf7a5680-8e38-11ea-8c33-ef5c89a7f0c7.png)

12. Scale down the browser window and confirm that the other logo is switched to at the same point the header gets a solid background):

![image](https://user-images.githubusercontent.com/177561/81029171-cf923600-8e38-11ea-9937-f3f9685ab4e1.png)

![image](https://user-images.githubusercontent.com/177561/81029156-c6a16480-8e38-11ea-9ded-f1569f222296.png)

13. Navigate to Customize > Header Settings > Appearance, and turn on the "Short Header".
14. View the Homepage, and subpages, and confirm the two header types are displaying as expected.

### Know issues
Copied from the comments below:

Depending on your exact settings, the 'alternative' logo might be needed for the Featured Image behind style, but might not work with the colour you have for the featured image beside option. If we run into that, I think the smartest thing to do might be to add a second logo upload, so you can pick a specific logo that will work for that background, too. It's not ideal -- a bit clunky UI-wise -- but I think it will be easier (and more predictable!) than trying to make the code 'smart' to try to guess the brightness of your logo. 

### Other information:
